AutoRaise.mm

```diff
@@ -334,6 +334,129 @@ - (void)onTick:(NSNumber *)timerInterval {
+NSArray   *parametersDictionary = @[@"delay", @"warpX", @"warpY", @"scale"];
+
+@interface ConfigClass:NSObject
+- (NSString *) getFilePath: (NSString *) filename;
+- (BOOL) fileExist: (NSString *) filename;
+- (void) readCommandLineConfig: (int) argc;
+- (void) readOriginalConfig;
+- (void) readHiddenConfig;
+- (void) validateParameters;
+@end
+
+@implementation ConfigClass
+
+- (NSString *) getFilePath: (NSString *) filename {
+    filename = [NSString stringWithFormat: @"%@/%@", NSHomeDirectory(), filename];
+    if (not [[NSFileManager defaultManager] fileExistsAtPath: filename]) { filename = @""; }
+    return filename;
+}
+
+- (BOOL) fileExist: (NSString *) filename {
+    filename = [NSString stringWithFormat: @"%@/%@", NSHomeDirectory(), filename];
+    return [[NSFileManager defaultManager] fileExistsAtPath: filename];
+}
+
+- (void) readCommandLineConfig: (int) argc {
+    if (argc > 1) {
+        // read NSArgumentDomain
+        NSUserDefaults *arguments = [NSUserDefaults standardUserDefaults];
+
+        for (id key in parametersDictionary) {
+            id arg = [arguments objectForKey: key];
+            if (arg != NULL) { parameters[key] = arg; }
+        }
+    } else {
+        [self readOriginalConfig];
+    }
+    return;
+}
+
+- (void) readOriginalConfig {
+    // original config files:
+    NSString *delayFile = @"AutoRaise.delay";
+    NSString *warpFile  = @"AutoRaise.warp";
+    
+    if ([self fileExist: delayFile] or [self fileExist: warpFile]) {
+        NSFileHandle *hDelayFile = [NSFileHandle fileHandleForReadingAtPath: [self getFilePath: delayFile]];
+        if (hDelayFile) {
+            parameters[@"delay"]= @(abs([[[NSString alloc] initWithData:
+                                           [hDelayFile readDataOfLength: 2] encoding:
+                                           NSUTF8StringEncoding] intValue]));
+            [hDelayFile closeFile];
+        }
+
+        NSFileHandle *hWarpFile = [NSFileHandle fileHandleForReadingAtPath: [self getFilePath: warpFile]];
+        if (hWarpFile) {
+            NSString *line = [[NSString alloc] initWithData:
+                [hWarpFile readDataOfLength:11] encoding:
+                NSUTF8StringEncoding];
+            NSArray *components = [line componentsSeparatedByString: @" "];
+            if (components.count >= 1) { parameters[@"warpX"] = @([[components objectAtIndex:0] floatValue]); }
+            if (components.count >= 2) { parameters[@"warpY"] = @([[components objectAtIndex:1] floatValue]); }
+            if (components.count >= 3) { parameters[@"scale"] = @([[components objectAtIndex:2] floatValue]); }
+            [hWarpFile closeFile];
+        }
+    } else {
+        [self readHiddenConfig];
+    }
+    return;
+}
+
+- (void) readHiddenConfig {
+    // search for dotfiles
+    NSString *hiddenConfigFile = @"";
+    if ([self fileExist: @".AutoRaise"]) { hiddenConfigFile = @".AutoRaise";
+    } else
```

Comment:
can be combined like so: } else if { ... }

Reply:
Ok